### ISSUE TITLE
vertical_whitespace_between_cases gives false warning if blank line has whitespace

### ISSUE THREAD

**Issue by BenStaveleyTaylor** [realm/SwiftLint#2538]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version: 0.29.2
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `vertical_whitespace_between_cases` rule flags a warning even when there is a blank line before the `case` or `default` keyword, if that blank line contains any white space. Totally whitespace lines should be accepted as being blank.

Explanation: Xcode tends to insert trailing whitespace when it formats text and its Text Editing preference for "Automatically trim trailing whitespace" is not reliable. We live in a world where there frequently is trailing whitespace and so we disable the `trailing_whitespace` rule.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ ./swiftlint --config swiftlint.yml
Loading configuration from 'swiftlint.yml'
Linting Swift files at paths 
Linting 'vertical_whitespace_between_cases.swift' (1/1)
/Users/bestave/Desktop/vertical_whitespace_between_cases/vertical_whitespace_between_cases.swift:9:1: warning: Vertical Whitespace Between Cases Violation: Include a single empty line between switch cases. (vertical_whitespace_between_cases)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.29.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods
* Paste your configuration file:

```yml
disabled_rules:
  - trailing_whitespace
  
opt_in_rules:
  - vertical_whitespace_between_cases
```

* Are you using nested configurations? No
* Which Xcode version are you using? 10.1
* Do you have a sample that shows the issue? 
```swift
enum Test {

    static func caseTest(value: Int) -> String {

        switch value {
        case 1:
            return "One"
         
        default:
            return "Not One"
        }
    }

}
```
Note: the blank line before `default` contains some trailing whitespace

**Comment by marcelofabri** [comment]
@BenStaveleyTaylor this should be a good starter bug if you wanna try fixing it!

**Comment by BenStaveleyTaylor** [comment]
@marcelofabri OK, that's a challenge then :-) I'll have a go shortly...

**Comment by marcelofabri** [comment]
Let me know if you have any questions! You should only have to change `VerticalWhitespaceBetweenCasesRule.swift`, so definitely start there.

**Comment by BenStaveleyTaylor** [comment]
@marcelofabri Thanks. I haven't been a contributor on any open source projects before so the process is my main concern. Are there some guidelines for new contributors I can read?

**Comment by marcelofabri** [comment]
We have some documentation in [CONTRIBUTING.MD](https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md), but feel free to ask any questions!
